tools/GraphClust/AlignCluster/align_cluster.xml

```diff
@@ -0,0 +1,230 @@
+          #if  str($genomic_cons_opts.genomic_cons_opts_selector) == "hg38_100way":
+             '$__tool_directory__/hg38.100way.nh'
+          #elif str($genomic_cons_opts.genomic_cons_opts_selector) == "hg38_20way":
+             '$__tool_directory__/hg38.20way.nh'
+          #else 
+         '$alignment_clustal'.ama $genomic_cons_opts.phylo_tree
+      #end if
+      &&
+    #else
+      touch alignment.evofold.out &&
+    #end if
+    
+    extract_conservation_metrics.py 
+    '$alignment_clustal' alignment.rnaz.out RESULTS/result.aln.sum alignment.evofold.out  
+    $cluster_all 
+    #if  str($bed_opts.bed_opts_selector) == "yes":
+        $bed_opts.transcript_loci_fasta 
+    '$bed_opts.transcript_loci_bed' 
+    '$bed_opts.genome_version' 
+  #else  
+    '' '' ''
+  #end if
+  conservation_metrics.tsv
+   
+]]>
+  </command>
+  <inputs>
+      <param type="data" name="cluster_all_fa" label="cluster-sequences-sorted" format="fasta"
+        help="Cluster sequences from the collect-result step"/>
+      <param type="data" name="cluster_all" label="cmsearch-results" format="txt"
+        help="Tabular cmsearch results of the cluster-collection step (CLUSTERS-cmsearch)"/>
+      <param name="results_top_num" type="integer" value="5" size="5" label="results_top_num" 
+        help="Top number of cmsearch hit sequences are used for structural alignment and metrics"/>
+
+      <conditional name="genomic_cons_opts">
+          <param type="select" name="genomic_cons_opts_selector" label="Phylo tree for evofold" 
+              help="For MAF genomic alignments as input data, corresponding phylo tree is required to perform Evofold2 structural conservation predcition. For other genomic alignemnts, please upload the tree file.">
+              <option value="no">No</option>
+              <option value="hg38_100way">hg38-100way</option>
+              <option value="hg38_20way">hg38-20way</option>
+              <option value="user_phylo_tree">from history</option>
+          </param>
+          <when value="no" />
+          <when value="hg38_100way" />
+          <when value="hg38_20way" />                
+          <when value="user_phylo_tree">
+              <param format="txt" name="phylo_tree" type="data" label="Newick phylo-tree" />
+          </when>
+      </conditional>
+      <conditional name="bed_opts">
+          <param type="select" name="bed_opts_selector" label="Extract genomic coordinates of the clusters" >
+              <option value="no">No</option>
+              <option value="yes">Yes</option>
+          </param>
+          <when value="no" />
+          <when value="yes" >
+              <param format="fasta" name="transcript_loci_fasta" type="data" label="Loci reference sequence"
+                help="Sequence of reference species (human) for the genomic loci (precursor mRNA/lncRNA) to locate the clusters" />
+              <param name='transcript_loci_bed' type="text" value="chr1 0 100000 gene 0 +" 
```

Comment:
You could also ask for 5 different fields ... chrom can be an integer? start, end integer ... name = string and strand boolean. score you could skip

Reply:
Chrom true. I would think better to stick to the bed format, it could be useful for some future extensions..

Reply:
What I mean is, instead of using a text field, collect all information with separate input fields and assemble this together to a `bed string`.

Reply:
Hmm, sorry not. I expected to receive email for the replies. Apparently not have received..
Let's have this one as future improvement :)